### ISSUE TITLE
Multi z-level vehicles exist

### ISSUE THREAD

**Issue by Procyonae** [CleverRaven/Cataclysm-DDA#67712]
### Describe the bug

So while messing with debug I saw an open sewer and figured I'd drive over it knowing non straight ramps are buggy and lost half my vehicle which while weird seemed fair enough.

![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/45432782/7b87742a-c165-4670-87e8-1eff1e3ed9bc)

Then I checked the level below to see the wreck it had left in the sewer... except instead it was still directly next to my vehicle, driving along with me despite being in solid ground

![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/45432782/e874aae0-2cb8-429a-9554-99b0cf1c8f16)

### Attach save file

[Tyringham-trimmed.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/12363983/Tyringham-trimmed.tar.gz)

### Steps to reproduce

Drive over an open sewer diagonally shearing your vehicle
Keep driving
Check the level below with clairvoyance and discover the rest of your vehicle still thinks it's attached

### Expected behavior

Idk not this

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.3324 (22H2)
- Game Version: 24b09f4 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth],
    No Rail Stations [No_Rail_Stations]
]

**Comment by Procyonae** [comment]
If you examine the vehicle it still shows the whole thing as being one btw

**Comment by Karol1223** [comment]
I think "multi-z-level" would work better than multitile here? Almost all vehicles are multitile, I think what you're working with here is multi-z-level

**Comment by Procyonae** [comment]
> I think "multi-z-level" would work better than multitile here? Almost all vehicles are multitile, I think what you're working with here is multi-z-level

🤦

**Comment by ZhilkinSerg** [comment]
Yeah, I am pretty sure while vehicles are 1 tile high,  vehicle parts can be positioned on different z-levels.

**Comment by rtxyd** [comment]
Is that possible to get like a two-levels bus in the future?

**Comment by Brambor** [comment]
Have center of mass needing to be within the ramp support:
```
Vehicle V with center of mass C going this way -> over Ramps R
VVVV R
VVCV
VVVV R
```
In this picture, the Center of mass will pass in between or over a ramp R. Since vehicles cannot turn on ramps, calculate it by projecting where the vehicle would go, if it went in a straight line. In case that does not happen, the vehicle would "tip over".

Tipping over sounds complicated, so let's just stop the player from being able to do that. By crashing into the ramp(s) (colliding with them).

This doesn't consider wheel position. That said, vehicles need only one tile under them to not fall and can drive over tips of trees when placed there [cannot find the bug report].

<hr>
Another solution I considered:
Check all vehicle tiles in the front. If some are on the ramp and some not, treat it as colliding with the ramp instead of driving on it.

**Comment by Brambor** [comment]
What happens IRL when vehicle goes onto a ramp partially: https://youtu.be/nvlU5c0D87s?t=10
